src/winkerberos.c

```diff
@@ -437,8 +437,136 @@ sspi_client_clean(PyObject* self, PyObject* args) {
     return Py_BuildValue("i", AUTH_GSS_COMPLETE);
 }
 
+#if PY_MAJOR_VERSION >= 3
+void destruct_channel_bindings_struct(PyObject* o) {
+    SecPkgContext_Bindings* context_bindings;
+    int offset = sizeof(SEC_CHANNEL_BINDINGS);
+    context_bindings = PyCapsule_GetPointer(o, NULL);
+#else
+void destruct_channel_bindings_struct(void* o) {
+    SecPkgContext_Bindings* context_bindings;
+    int offset = sizeof(SEC_CHANNEL_BINDINGS);
+    context_bindings = (SecPkgContext_Bindings *)o;
+#endif
+
+    if (context_bindings != NULL) {
+        free(context_bindings);
+    }
+}
+
+PyDoc_STRVAR(sspi_build_channel_bindings_struct_doc,
+"buildChannelBindingsStruct(**kwargs)\n"
+"\n"
+"Builds a SecPkgContext_Bindings struct and returns a pointer to this struct.\n"
+"This struct can then be passed into `authGSSClientStep` under the\n"
+"input_chan_bindings kwarg argument. The SecPkgContext_bindings object\n"
+"destroys itself when it is reclaimed. While the parameters supported by this\n"
+"method match the structure of the GSSAPI gss_channel_bindings_t\n"
+"https://docs.oracle.com/cd/E19455-01/806-3814/overview-52/index.html\n"
+"Windows only uses the application_data parameter for now. The other\n"
+"parameters are kept for compatibility with ccs-kerberos and future\n"
+"compatibility.\n"
+"\n"
+":Parameters:\n"
+"  - `initiator_addrtype`: Optional int of the initiator address type,\n"
+"    default is GSS_C_AF_UNSPEC (0).\n"
+"  - `initiator_address`: Optional byte string of the initiator address,\n"
+"    leave blank for SSPI operations.\n"
+"  - `acceptor_addrtype`: Optional int of the acceptor address type,\n"
+"    default is GSS_C_AF_UNSPEC (0).\n"
+"  - `acceptor_address`: Optional byte string of the acceptor address,\n"
+"    leave blank for SSPI operations.\n"
+"  - `application_data`: Optional byte string that should be set under the\n"
+"    application_data field. An example of this would be\n"
+"    'tls-server-endpoint:{cert-hash}' where {cert-hash} is the hash of the\n"
+"    server's certificate.\n"
+"\n"
+":Returns: A tuple of (result, SecPkgContext_Bindings*) where result is the\n"
+"result code and SecPkgContext_Bindings is the channel bindings structure\n"
+"that can be passed onto L{authGSSClientStep}.");
+static PyObject*
+sspi_build_channel_bindings_struct(PyObject* self, PyObject* args, PyObject* keywds) {
+    static char *kwlist[] = {"initiator_addrtype", "initiator_address", "acceptor_addrtype",
+        "acceptor_address", "application_data", NULL};
+    SEC_CHANNEL_BINDINGS* channel_bindings;
+    SecPkgContext_Bindings* context_bindings;
+    PyObject *pychan_bindings = NULL;
+    int result = 0;
+    int initiator_addrtype = 0;
+    int acceptor_addrtype = 0;
+    int offset = 0;
+    int data_length = 0;
```

Comment:
Since you're adding three ints together you need more than an int to avoid overflow. Since this value is going to passed to malloc, use size_t.